pkg/plugins/runtime/gateway/listener_generator.go

```diff
@@ -54,11 +54,14 @@ func GenerateListener(info GatewayListenerInfo) (*envoy_listeners.ListenerBuilde
 	}
 
 	// TODO(jpeach) if proxy protocol is enabled, add the proxy protocol listener filter.
-	return envoy_listeners.NewListenerBuilder(info.Proxy.APIVersion).
+	return envoy_listeners.NewInboundListenerBuilder(
+		info.Proxy.APIVersion,
+		address,
+		port,
+		core_xds.SocketAddressProtocolTCP,
+	).
+	WithOverwriteName(name).
```

Comment:
Ah, sorry - one oversight
```suggestion
		WithOverwriteName(name).
```